### DISCUSSION TITLE
[Arcropolis 3.4.0] [CSS crash + software aborting] on 4 player smash

### DISCUSSION THREAD

**Discussion by Skuuurt** [Raytwo/ARCropolis#351] [Issues] [answered]
System Firmware Version
14.1.1

Atmosphere Version
1.3.2

Smash Version
13.0.1

ARCropolis Version
3.4.0

Details

I am experiencing random css crashes with only 4 skin mods in the mods folder. 

Two skins have just been reslotted to my likings, that's the only modification I made after downloading the 4 of them (the pirate Funky Kong and the Dread suit Samus ones), but the submitters of these 2 reslotted mods said it was ok on their respective pages, plus leaving them on their original spot does not change anything. The changes have been made accordingly for these 2 skins (3 skins out of 4 use a config file). 
You can verify it yourself in the archive provided. That's the archive I'm using in the video.

Using latest arcropolis 3.4.0 Ray provided on the smash ultimate modding discord the 24/05/2022 on this message : 
https://discord.com/channels/447823426061860879/447940880922574848/978478366816604240

Also tested with latest official stable 3.4.0, the results are the same.

No use of any other plugin than libarcropolis.nro
SD card is formatted to fat32

 Those crashes do not always occur right away, sometimes I have to select the skins again 2, 3, 4, 5 times more and so on... But one thing is sure, the larger the modpack is, the more often and quickly it will occur. 

Log : 
[01653670075_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8788984/01653670075_01006a800016e000.log)

Additional log by #1morbhead : 
[01653672362_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8789035/01653672362_01006a800016e000.log)

Files used : 
https://cdn.discordapp.com/attachments/506286717519659023/979795028446679150/mods.part1.rar
https://cdn.discordapp.com/attachments/506286717519659023/979795099506597898/mods.part2.rar

Video of the crash correponding to the first log above : 
https://cdn.discordapp.com/attachments/506286717519659023/979794778952716418/Random_CSS_crashes.mp4

Original links of the mods in case you'd like to try from scratch : 

- Pirate Funky Kong (reslotted, was on c03 on the downloaded version) : https://gamebanana.com/mods/357706
- Metroid suit samus (2nd link on the page, reslotted, was on c07 on the downloaded version) : https://gamebanana.com/mods/326774
- Safer Sephiroth (1st download link on the page,  named "default alt 1", got rid of c06 and only kept c07, not reslotted) : https://gamebanana.com/mods/358758
- Devil Jin (not reslotted, kept it at c06 like it was on the link) : https://gamebanana.com/mods/373189

**Comment by marcym2** [comment]
Test it without any mods on, it will probably happen as well.

 I've been trying to diagnose what was the cause of these crashes with my mods one by one, but my experience has been the same as yours so far. It simply crashes randomly when playing with more than 2 players, and even reverting to picking 4 players the game freezes as well.
 
 Sorry for not answering your question , but I found myself stumped with this so atleast someone else is having this specific problem that arose recently.

I even tried going to elite smash and it froze in the waiting lobby. Hopefully it will get figured out.

**Reply by Raytwo** [reply]
We're totally aware of the issue, but it's super hard to pinpoint for some reasons.
A complete rewrite of ARCropolis is ongoing, but this will take a while and while I'm hopeful the problem will "fix itself" by reviewing basically the entirety of the code, I have no guarantee that will be the case.
It's kind of difficult determining if it's worth going all out trying to figure out the issue on the current codebase that's honestly very hard to read and maintain or just try rewriting everything and hopefully catching the issue on the way.

**Reply by Skuuurt** [reply]
As far as I know, I never had this type of crash with no mods installed. Plus I noticed the problem I described generally happens with large file mods. As a consequence, it seems that reworking the textures of said mods would be a good temporary fix while waiting the necessary time for the rewrite to be done. And then check if the rewrite changes something on that, which as Raytwo mentionned will not be guaranteed.
As an example, I lightened the textures of two of the 4 mods mentionned above, and experienced no css crash since.
So at least and for the time being, it is no more an unsolvable issue if you are willing to put you hands on the larger textures that causes it to happen.

**Comment by marcym2** [comment]
Thanks for the clarification! Hopefully all goes well with the code rewrite.

**Comment by marcym2** [comment]
I tried having every single mod disabled in the acropolis mod manager
in-game and I still experienced a crash when using 3 or more players. I can
test more use-cases if you'd like.

**Reply by Skuuurt** [reply]
Try to not only disable them, but also remove them entirely from the mods folder. Plus if you have any other plugin than libarcropolis.nro in your skyline folder, they should be put off that folder too. 3rd thing when you have done all that, try to reinstall arcropolis 3.4.0 from scratch and retry. A video to showcase your issue could also be a good thing to witness it.

I highly doubt this is the same problem that the one I had, as I highly doubt latest stable Arcropolis could crash without any mod installed. And as I said, in my case the cause of the issue was mainly the skins I was using.

**Comment by ngodanie** [comment]
I've been experiencing the exact same issue, except with even just 2-player smash. I haven't had any issues until I recently installed Training Modpack 4.0.1. I don't know how it will react when I revert the Training Modpack, but that is my next troubleshooting step.

Screenshot:
![01654222075_01006a800016e000](https://user-images.githubusercontent.com/101282287/171774786-f7c6c822-4d1e-4f7f-a9cc-a366bc82b3a0.jpg)

Log:
[01654222075_01006a800016e000.log](https://github.com/Raytwo/ARCropolis/files/8829077/01654222075_01006a800016e000.log)

**Reply by Skuuurt** [reply]
So if you read carefully what I wrote, your issue is not the "exact same issue" since you have different conditions and moreover, an installed plugin in your skyline folder. 

Please stop writing on this discussion and open your own after you checked on discord with the devs if it is okay to do so, as it is (again) probably not related to the issue I had. This won't help the devs at all, since your issues are different.

Actually, in your case you should have started with uninstalling the training modpack in the 1st place, as it is well-known that some plugins could currently cause some issues and increase instability.

**Comment by Raytwo** [comment, marked as answer]
Issue has been fixed as of ARCropolis 3.5.0 (coming out today, or tonight in Skuuurt time)
